.travis.yml

```diff
@@ -1,80 +1,79 @@
-        - mvn test -Dtest=IngestionTests -B
-    - stage: integration tests
+    - stage: tests
+      name: "Integration Tests"
+      services: docker
       before_script:
+        - pip3 install -r requirements.txt
+        - git clone https://github.com/bats-core/bats-core.git
+        - pushd bats-core && git checkout v1.1.0 && sudo ./install.sh /usr/local && popd
         - cd deploy
-        - make bootstrap ARGS='--inbox mina --keyserver ega'
+        - make prepare
+        - make bootstrap ARGS="--keyserver ega --with-fake-cega"
         - sudo chown -R travis private
-        - docker network create cega
-        - make up
-        - docker-compose ps
+        - make up OMIT="--scale res=0"
+        - make preflight-check
       script:
-        - cd tests
-        - mvn test -Dtest=OutgestionTests -B
-    - stage: integration tests
+        - cd ../tests
+        - bats integration
+    - stage: tests
+      name: "Robustness Tests"
+      services: docker
       before_script:
+        - pip3 install -r requirements.txt
+        - git clone https://github.com/bats-core/bats-core.git
+        - pushd bats-core && git checkout v1.1.0 && sudo ./install.sh /usr/local && popd
         - cd deploy
-        - make bootstrap ARGS='--inbox mina --keyserver ega'
+        - make prepare
+        - make bootstrap ARGS="--keyserver ega --with-fake-cega"
         - sudo chown -R travis private
-        - docker network create cega
-        - make up
-        - docker-compose ps
+        - make up OMIT="--scale res=0"
+        - make preflight-check
       script:
-        - cd tests
-        - mvn test -Dtest=RobustnessTests -B
-    - stage: integration tests
+        - cd ../tests
+        - bats robustness
+    - stage: tests
+      name: "Stress Tests"
+      services: docker
+      script: echo "Stress Tests.... hmm... later"
+    - stage: tests
+      name: "Security Tests"
+      services: docker
       before_script:
+        - pip3 install -r requirements.txt
+        - git clone https://github.com/bats-core/bats-core.git
+        - pushd bats-core && git checkout v1.1.0 && sudo ./install.sh /usr/local && popd
         - cd deploy
-        - make bootstrap
+        - make prepare
+        - make bootstrap ARGS="--keyserver ega --with-fake-cega"
         - sudo chown -R travis private
-        - docker network create cega
-        - docker-compose up -d #--scale ingest=3 --scale verify=5
-        - docker-compose ps
+        - make up OMIT="--scale res=0"
+        - make preflight-check
       script:
-        - sleep 5
-        - pip install -r https://raw.githubusercontent.com/NBISweden/LocalEGA-cryptor/master/requirements.txt
-        - pip install git+https://github.com/NBISweden/LocalEGA-cryptor.git
-        - pip install pika
-        - make -C test
-        - sleep 10
-        - make -C test check
+        - cd ../tests
+        - bats security
+    - stage: images
+      name: "Images Build"
+      script: echo 'Next PR'
```

Comment:
I assume Image build will be part of a different PR ?

Reply:
Correct. 
I have a description in the presentation for tomorrow, and I'm involving the team to take a decision.